### PR TITLE
  Add/tidy API doxygen for the new pico_i2c_slave library

### PR DESCRIPTION
**Pull request by andygpz11**
Add/tidy API doxygen for the new pico_i2c_slave library (see PR #1205)

### CODE REVIEW

**Review by andygpz11 — COMMENTED** [review]
Thanks, LGTM.
I think we need another reviewer at this point :)

**Comment by kilograham** [comment]
yeah, i can review it sill, but need to actually generate the docs to check how the link works

File: src/rp2_common/pico_i2c_slave/include/pico/i2c_slave.h [hunk 1]
```diff
@@ -39,12 +39,13 @@ typedef enum i2c_slave_event_t
...
  * \brief I2C slave event handler
+ * \ingroup pico_i2c_slave
  * 
  * The event handler will run from the I2C ISR, so it should return quickly (under 25 us at 400 kb/s).
  * Avoid blocking inside the handler and split large data transfers across multiple calls for best results.
- * When sending data to master, up to `i2c_get_write_available()` bytes can be written without blocking.
- * When receiving data from master, up to `i2c_get_read_available()` bytes can be read without blocking.
- * \ingroup pico_i2c_slave
+ * When sending data to master, up to \ref i2c_get_write_available()  bytes can be written without blocking.
+ * When receiving data from master, up to `\ref 2c_get_read_available() bytes can be read without blocking.
```
Comment (peterharperuk) [inline src/rp2_common/pico_i2c_slave/include/pico/i2c_slave.h:47]: is that a rogue backquote?

Comment (andygpz11) [reply src/rp2_common/pico_i2c_slave/include/pico/i2c_slave.h:47]: yes

**Review by kilograham — APPROVED** [review]

**Pull request merged**